pkg/metadata/inventories/inventories.go

```diff
@@ -54,22 +59,46 @@ var (
 	timeSince = time.Since
 )
 
+// AgentMetadataName is an enum type containing all defined keys for
+// SetAgentMetadata.
+type AgentMetadataName string
+
+// Constants for the metadata names; these are defined in
+// pkg/metadata/inventories/README.md and any additions should
+// be updated there as well.
 const (
-	// CloudProviderMetatadaName is the field name to use to set the cloud
-	// provider name in the agent metadata.
-	CloudProviderMetatadaName = "cloud_provider"
-	// HostnameSourceMetadataName is the field name to use to set the hostname
-	// source in the agent metadata.
-	HostnameSourceMetadataName = "hostname_source"
+	AgentCloudProvider                AgentMetadataName = "cloud_provider"
+	AgentHostnameSource               AgentMetadataName = "hostname_source"
+	AgentVersion                      AgentMetadataName = "agent_version"
+	AgentFlavor                       AgentMetadataName = "flavor"
+	AgentConfigApmDDURL               AgentMetadataName = "config_apm_dd_url"
+	AgentConfigDDURL                  AgentMetadataName = "config_dd_url"
+	AgentConfigSite                   AgentMetadataName = "config_site"
+	AgentConfigLogsDDURL              AgentMetadataName = "config_logs_dd_url"
+	AgentConfigLogsSocks5ProxyAddress AgentMetadataName = "config_logs_socks5_proxy_address"
+	AgentConfigNoProxy                AgentMetadataName = "config_no_proxy"
+	AgentConfigProcessDDURL           AgentMetadataName = "config_process_dd_url"
+	AgentConfigProxyHTTP              AgentMetadataName = "config_proxy_http"
+	AgentConfigProxyHTTPS             AgentMetadataName = "config_proxy_https"
+	AgentInstallerVersion             AgentMetadataName = "install_method_installer_version"
+	AgentInstallTool                  AgentMetadataName = "install_method_tool"
+	AgentInstallToolVersion           AgentMetadataName = "install_method_tool_version"
```

Comment:
```suggestion
	AgentInstallMethodInstallerVersion             AgentMetadataName = "install_method_installer_version"
	AgentInstallMethodTool                  AgentMetadataName = "install_method_tool"
	AgentInstallMethodToolVersion           AgentMetadataName = "install_method_tool_version"
```
Let's make sure the InitialCaps Go identifier and the snake_case key match.